### PR TITLE
Docs: Overhaul the default keys list

### PR DESCRIPTION
**Pull request by Firehawke**
I'm expecting some need for refinement, so I'll leave this open for the time being.

### CODE REVIEW

File: docs/source/usingmame/defaultkeys.rst [hunk 1]
```diff
@@ -1,106 +1,476 @@
...
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **P**                              | Player 1 Mahjong/Hanafuda P                                                                               |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **Q**                              | Player 1 Mahjong/Hanafuda Q                                                                               |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **Right Alt**                      | Player 1 Mahjong/Hanafuda Last Chance                                                                     |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **Right Ctrl**                     | Player 1 Mahjong/Hanafuda Score                                                                           |
```
Comment (cuavas) [inline docs/source/usingmame/defaultkeys.rst:361]: This should say “Take Score”, not just “Score”.  The source should be changed as well.

File: docs/source/usingmame/defaultkeys.rst [hunk 2]
```diff
@@ -1,106 +1,476 @@
...
+| **W**                              | Key Out                                                                                                   |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **S**                              | Low                                                                                                       |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **9**                              | Service                                                                                                   |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **0**                              | Book-Keeping                                                                                              |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
```
Comment (cuavas) [inline docs/source/usingmame/defaultkeys.rst:435]: The poker and slots controls should be listed in this section as well.

File: docs/source/usingmame/defaultkeys.rst [hunk 3]
```diff
@@ -1,106 +1,476 @@
...
+|                                    | | All emulations which require emulated keyboards will start in that mode and you can only access         |
+|                                    | | the internal UI (hitting TAB) by first hitting this key. You can change the initial status of the       |
+|                                    | | emulated keyboard as presented upon start by using **-ui_active**.                                      |
+|                                    |                                                                                                           |
+|                                    | FIXME: Add link back to -ui_active                                                                        |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **Escape**                         | Exits emulator.                                                                                           |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
```
Comment (cuavas) [inline docs/source/usingmame/defaultkeys.rst:166]: Are you going to describe the other UI controls?  For example the navigation directions (used on most menus), focus next/previous (used on the system and software selection menus), and the other controls used on the system/software selection menus?

File: docs/source/usingmame/defaultkeys.rst [hunk 4]
```diff
@@ -1,106 +1,476 @@
...
+|                                    | | All emulations which require emulated keyboards will start in that mode and you can only access         |
+|                                    | | the internal UI (hitting TAB) by first hitting this key. You can change the initial status of the       |
+|                                    | | emulated keyboard as presented upon start by using **-ui_active**.                                      |
+|                                    |                                                                                                           |
+|                                    | FIXME: Add link back to -ui_active                                                                        |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **Alphanumeric Keys**              | These keys are mapped to their equivalents in the emulated machine by default.                            |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
```
Comment (cuavas) [inline docs/source/usingmame/defaultkeys.rst:464]: Tape controls probably belong in this section, and the UI paste control.

File: docs/source/usingmame/defaultkeys.rst [hunk 5]
```diff
@@ -1,106 +1,476 @@
...
+to make it easier to find what you're looking for.
+
+All of the controls below are fully configurable in the user interface. These 
+charts show the default configuration.
+
+MAME Controls
+=============
+
+The controls here cover MAME functions such as MAME's menus, machine pause,
+and saving/loading save states.
```
[the remark this answers is not in the document]
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:19]: Will be commenting on each as I prep them for review.
Added.

File: docs/source/usingmame/defaultkeys.rst [hunk 6]
```diff
@@ -1,106 +1,476 @@
...
+| **Shift+~**                        | While paused, loads the most recent rewind save state.                                                    |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **-** *(not numeric keypad)*       | Volume Down for machines that have a volume control.                                                      |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **=** *(not numeric keypad)*       | Volume Up for machines that have a volume control.                                                        |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **F1**                             | | Power the machine on for machines that have specific power button behavior.                             |
+|                                    | | Memory reset button for machines that have a button for clearing NVRAM, typically gambling machines.    |
```
[the remark this answers is not in the document]
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:57]: Corrected.

File: docs/source/usingmame/defaultkeys.rst [hunk 7]
```diff
@@ -1,106 +1,476 @@
...
+| **F3**                             | Resets the machine.                                                                                       |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **Shift+F3**                       | | Performs a "hard reset", which tears everything down and re-creates it from scratch.                    |
+|                                    | | This is a more thorough and complete reset than the reset you get from hitting F3                       |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **LCtrl+F3**                       | Toggle uneven stretch. (*SDL ONLY*)                                                                       |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **F4**                             | | Shows the game palette, decoded GFX, and any tilemaps. Use the Enter key to                             |
```
[the remark this answers is not in the document]
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:69]: This is a copy-paste from the original docs. Corrected.

File: docs/source/usingmame/defaultkeys.rst [hunk 8]
```diff
@@ -1,106 +1,476 @@
...
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **F6**                             | Toggle cheat mode. (if started with "-cheat")                                                             |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **LCtrl+F6**                       | Decrease Prescaling.                                                                                      |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **LCtrl+F7**                       | Increase Prescaling.                                                                                      |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **F7**                             | | Load a save state. You will be requested to press a key to determine which save state you wish          |
```
[the remark this answers is not in the document]
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:123]: Adjusted along with the save section.

File: docs/source/usingmame/defaultkeys.rst [hunk 9]
```diff
@@ -1,106 +1,476 @@
...
+|                                    | | (*Windows non-SDL MAME only*)                                                                           |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **Page DN**                        | | Fast forward. While held, runs game with throttling disabled and with the maximum frameskip.            |
+|                                    | | (*SDL MAME only*)                                                                                       |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **Alt+ENTER**                      | Toggles between full-screen and windowed mode.                                                            |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **Scroll Lock**                    | Default mapping for the **uimodekey**.                                                                    |
```
[the remark this answers is not in the document]
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:156]: Added.

File: docs/source/usingmame/defaultkeys.rst [hunk 10]
```diff
@@ -1,106 +1,476 @@
...
+|                                    |                                                                                                           |
+|                                    | | This key allows users to disable and enable the emulated keyboard in machines that require it.          |
+|                                    | | All emulations which require emulated keyboards will start in that mode and you can only access         |
+|                                    | | the internal UI (hitting TAB) by first hitting this key. You can change the initial status of the       |
+|                                    | | emulated keyboard as presented upon start by using **-ui_active**.                                      |
```
[the remark this answers is not in the document]
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:161]: This is from the original docs as well. It's been adjusted somewhat and we can talk more on this subject as needed.

File: docs/source/usingmame/defaultkeys.rst [hunk 11]
```diff
@@ -1,106 +1,476 @@
...
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **2**                              | Player 2 start or 2 players mode                                                                          |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **3**                              | Player 3 start or 3 players mode                                                                          |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **4**                              | Player 4 start or 4 players mode                                                                          |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **Backspace**                      | Bill 1 FIXME: Bill 1 does?                                                                                |
```
[the remark this answers is not in the document]
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:195]: Corrected. I had intended to get that before I did the PR, but missed that one.

File: docs/source/usingmame/defaultkeys.rst [hunk 12]
```diff
@@ -1,106 +1,476 @@
...
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **9**                              | Service 1 (Service buttons are typically used to give free credits)                                       |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **0**                              | Service 2                                                                                                 |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **-** (not numeric keypad)         | Service 3                                                                                                 |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **=** (not numeric keypad)         | Service 4                                                                                                 |
```
[the remark this answers is not in the document]
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:203]: Corrected.

File: docs/source/usingmame/defaultkeys.rst [hunk 13]
```diff
@@ -1,106 +1,476 @@
...
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **N**                              | Player 1 Mahjong/Hanafuda N                                                                               |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **O**                              | Player 1 Mahjong/Hanafuda O                                                                               |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **P**                              | Player 1 Mahjong/Hanafuda P                                                                               |
++------------------------------------+-----------------------------------------------------------------------------------------------------------+
+| **Q**                              | Player 1 Mahjong/Hanafuda Q                                                                               |
```
[the remark this answers is not in the document]
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:357]: Adjusted.

File: docs/source/usingmame/defaultkeys.rst (hunk 1 above)
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:361]: Corrected in docs and code.

File: docs/source/usingmame/defaultkeys.rst (hunk 2 above)
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:435]: Added.

File: docs/source/usingmame/defaultkeys.rst (hunk 3 above)
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:166]: Covered now for the most part. There are a few elements that aren't obvious yet though. We'll talk about those when I submit the next push.

File: docs/source/usingmame/defaultkeys.rst (hunk 4 above)
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:464]: Got those now as well.

File: docs/source/usingmame/defaultkeys.rst [hunk 14]
```diff
@@ -1,106 +1,558 @@
...
+
+    Use the Enter key to switch between the three modes (palette, graphics, and
+    tilemaps).
+
+    Press F4 again to turn off the display.
+    The key controls in each mode vary slightly:
+
+    Palette/colortable mode:
+
+    * **[** **]** - switch between palette and colortable modes.
```
Comment (cuavas) [inline docs/source/usingmame/defaultkeys.rst:110]: Should be “switch between palette devices” or something – I think it’s one per `device_palette_interface` in the system.

File: docs/source/usingmame/defaultkeys.rst [hunk 15]
```diff
@@ -1,106 +1,558 @@
...
+**LCtrl+F6**
+    Decrease Prescaling.
+**LCtrl+F7**
+    Increase Prescaling.
+**F7**
+    Load a save state. You will be prompted to press a key or select from the
+    menu to determine which save state you wish to load.
+
+    *Note that the save state feature is not supported for a large number of
+    drivers. If support is not enabled for a given driver, you will receive a
```
Comment (cuavas) [inline docs/source/usingmame/defaultkeys.rst:159]: This is unclear – it’s always enabled, it’s just not necessarily supported/known to work.  You get the warning if it isn’t marked as expected to work.

File: docs/source/usingmame/defaultkeys.rst [hunk 16]
```diff
@@ -1,106 +1,558 @@
...
+    Service 2
+**-** (not numeric keypad)
+    Service 3
+**=** (not numeric keypad)
+    Service 4
+**Ctrl**/**Mouse B0**/**Gun 1 Button 0**
+    Mahjong Kan
+**Alt**/**Mouse B2**/**Gun 1 Button 1**
+    Mahjong Pon
+**Spacebar**/**Mouse B1**/**Joystick 1 Button 1**
```
Comment (cuavas) [inline docs/source/usingmame/defaultkeys.rst:348]: Are you actually getting joystick/mouse/gun defaults on these?  From a quick look at the source, it’s only got the keyboard inputs there.  A quick run of `hotgmck` seemed to confirm this.

File: docs/source/usingmame/defaultkeys.rst [hunk 17]
```diff
@@ -1,106 +1,558 @@
...
+**I**
+    Payout
+**O**
+    Door
+**Q**
+    Key In
+**W**
+    Key Out
+**S**
+    Low
```
Comment (cuavas) [inline docs/source/usingmame/defaultkeys.rst:460]: These could be organised better.  I’d try to keep related keys together.
* Bonus game control Take Score, Double Up, High, Low should be together.
* Payout should Key In and Key Out should probably come immediately after the coin and bill controls.
* Bet and Half Gamble probably belong near Start.

Comment (cuavas) [reply docs/source/usingmame/defaultkeys.rst:460]: Deal and Stand are blackjack controls, although Deal is also used for poker.  They should probably be together.

File: docs/source/usingmame/defaultkeys.rst [hunk 18]
```diff
@@ -1,106 +1,558 @@
...
+**C**
+    Hold 3
+**V**
+    Hold 4
+**B**
+    Hold 5
+**N**
+    Cancel
+**1**
+    Bet
```
Comment (cuavas) [inline docs/source/usingmame/defaultkeys.rst:488]: Not related to this directly, but why do we even have this?  It conflicts with the Start key.  We should probably just get rid of it and make poker games use the regular Bet input.

File: docs/source/usingmame/defaultkeys.rst (hunk 14 above)
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:110]: This isn't a tool I'm familiar with, so I'll take your word on it. Adjusted.

File: docs/source/usingmame/defaultkeys.rst (hunk 15 above)
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:159]: Adjusted. You may want further adjustments on this one.

File: docs/source/usingmame/defaultkeys.rst (hunk 16 above)
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:348]: Oddly, it did once with a controller file that didn't map any Mahjong buttons when I first loaded one up to start pulling mappings, but I can't reproduce it with a completely clean set.

I'll remove those shortly.

File: docs/source/usingmame/defaultkeys.rst (hunk 17 above)
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:460]: I've adjusted this a bit. While I know how to play blackjack and poker, I'm not acquainted with the gambling machines so I've reworked them on best knowledge and they may need some further adjustments (if, e.g. they don't typically have the double up, high, low, etc)

File: docs/source/usingmame/defaultkeys.rst (hunk 18 above)
Comment (Firehawke) [reply docs/source/usingmame/defaultkeys.rst:488]: Great question that I don't have a great answer for. If this is going to happen, let me know so I can do the adjustments needed for the docs.

**Comment by cuavas** [comment]
I’m going to merge this now because it’s already a big improvement over what we have.  I might rearrange the mahjong and gambling controls a bit myself.  Speaking of which, is it worth having a general “gambling” section in addition to the specific blackjack, poker and slots sections, or should the game-specific controls be merged with the general gambling controls section?

**Comment by cuavas** [comment]
Oh, one other thing – the title of the page picks up the first top-level heading, so the title shown in window names, etc. is “Controls Foreword” now.

**Pull request merged**